controllers/apply/lib/field-types/field.js

```diff
@@ -19,34 +17,77 @@ class Field {
             throw new Error('Must provide name');
         }
 
-        this.label = props.label;
+        const label = props.label ? props.label : this.defaultLabel();
+        if (label) {
+            this.label = label;
+        } else {
+            // throw new Error('Must provide label');
+        }
+
         this.labelDetails = props.labelDetails;
         this.explanation = props.explanation;
 
-        this.type = props.type || 'text';
+        this.type = props.type ? props.type : this.getType();
+
+        this.attributes = defaults(
+            { size: 30 },
+            this.defaultAttributes(),
+            props.attributes
+        );
 
-        this.attributes = defaults({ size: 30 }, props.attributes);
         this.settings = props.settings || {};
 
         this.isRequired = props.isRequired !== false;
 
-        if (props.schema) {
-            this.schema = props.schema;
-        } else {
-            this.schema = this.isRequired
-                ? Joi.string().required()
-                : Joi.string()
-                      .allow('')
-                      .optional();
-        }
+        this.schema = props.schema ? props.schema : this.defaultSchema();
 
-        this.messages = props.messages || [];
+        // @TODO Should this merge based on key rather than a plain concat?
+        this.messages = this.defaultMessages().concat(props.messages || []);
 
         this.value = undefined;
         this.errors = [];
         this.featuredErrors = [];
     }
 
+    getType() {
+        return 'text';
+    }
+
+    /**
+     * Allow sub-classes to provide a default label
+     */
+    defaultLabel() {
+        return null;
+    }
+
+    /**
+     * Allow sub-classes to provide a default schema
+     * also allows it to be accessed by the instance
+     */
+    defaultSchema() {
```

Comment:
Added these `default*` methods to make it easier for the sub-types to define their own without having to duplicate as much boilerplate in the constructor. See `phone.js` for a good example of this.